File: 프로그래머스/1/12935. 제일 작은 수 제거하기/제일 작은 수 제거하기.js
function solution(arr) {
    var answer = [];
    
    for(let i=0;i<arr.length;i++){
        if(arr[i] === Math.min(...arr))
            continue;
        
        answer.push(arr[i]);
    }
    
    if(answer.length === 0)
        answer.push(-1);
    
    return answer;
}

File: 프로그래머스/1/12935. 제일 작은 수 제거하기/제일 작은 수 제거하기.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool comp(int a, int b){
    return a>b;
}

vector<int> solution(vector<int> arr) {
    vector<int> answer;
    
    for(int i=0;i<arr.size();i++)
        answer.push_back(arr[i]);
    
    int min_index = min_element(answer.begin(), answer.end()) - answer.begin();
    answer.erase(answer.begin() + min_index);

    if(answer.empty())
        answer.push_back(-1);
    
    return answer;
}